src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -23,10 +24,112 @@ private static string GetGitSystemHash(GitPreparer gitPreparer, IFileSystem file
+                // to ignore the exception and continue enumerating the remaining files and 
+                // folders. It is also possible (but unlikely) that a DirectoryNotFound exception 
+                // will be raised. This will happen if currentDir has been deleted by
+                // another application or thread after our call to Directory.Exists. The 
+                // choice of which exceptions to catch depends entirely on the specific task 
+                // you are intending to perform and also on how much you know with certainty 
+                // about the systems on which this code will run.
+                catch (UnauthorizedAccessException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+                catch (System.IO.DirectoryNotFoundException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+
+                string[] files = null;
+                try
+                {
+                    files = System.IO.Directory.GetFiles(currentDir);
+                }
+
+                catch (UnauthorizedAccessException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+
+                catch (System.IO.DirectoryNotFoundException e)
+                {
+                    Logger.WriteError(e.Message);
+                    continue;
+                }
+                // Perform the required action on each file here.
+                // Modify this block to perform your required task.
+                foreach (string file in files)
+                {
+                    try
+                    {
+                        // Perform whatever action is required in your scenario.
+                        System.IO.FileInfo fi = new System.IO.FileInfo(file);
+                        result.Add(fi.Name);
+
+                        var reader = fi.OpenText();
+                        result.Add(reader.ReadToEnd());
+
+                        //Logger.WriteInfo(string.Format("{0}: {1}, {2}", fi.Name, fi.Length, fi.CreationTime));
+                    }
+                    catch (System.IO.FileNotFoundException e)
+                    {
+                        // If file was deleted by a separate application
+                        //  or thread since the call to TraverseTree()
+                        // then just continue.
+                        Logger.WriteError(e.Message);
+                        continue;
+                    }
+                }
+
+                // Push the subdirectories onto the stack for traversal.
+                // This could also be done before handing the files.
+                // push in reverse order
+                for (int i = subDirs.Length - 1; i >= 0; i--)
+                {
+                    dirs.Push(subDirs[i]);
```

Comment:
Yes, there will be a traversal error. The current directory does not change anywhere, so it will never go into the subdirectories.